helix-view/src/editor.rs

```diff
@@ -286,10 +286,35 @@ pub struct Config {
     pub soft_wrap: SoftWrap,
     /// Workspace specific lsp ceiling dirs
     pub workspace_lsp_roots: Vec<PathBuf>,
+    /// Contextual information on top of the viewport
+    pub sticky_context: StickyContextConfig,
     /// Which line ending to choose for new documents. Defaults to `native`. i.e. `crlf` on Windows, otherwise `lf`.
     pub default_line_ending: LineEndingConfig,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
+#[serde(default, rename_all = "kebab-case", deny_unknown_fields)]
+pub struct StickyContextConfig {
+    /// Display context of current top view if it is outside the view. Default to off
+    pub enable: bool,
+
+    /// Display an indicator whether to indicate if the sticky context is active
+    /// Eventually making this a string so that it is configurable.
+    pub indicator: bool,
+
+    /// The max amount of lines to be displayed. (including indicator!)
+    /// The viewport is taken into account when changing this value.
+    /// So if the configured amount is more than the viewport height, it will be capped to a max
+    /// of the complete viewport height.
+    ///
+    /// Default: 0, which means that it is a fixed size based on the viewport
+    pub max_lines: u16,
```

Comment:
I think this should be capped to a third of the viewport height. We never want to overlay the entire viewport IMO (and in general more than a third seems too much). This is also more consistent that way `max_context_height = min(view_height/3, max_lines)` without a special case for `0`.

 I also think a more aggressive default would be good. Maybe 10! More than 10 lines seems excessive (if people prefer a larger value they can always increase this furthers)